static/elements/chromedash-upcoming-milestone-card.js

```diff
@@ -0,0 +1,170 @@
+import {LitElement, html} from 'lit-element';
+import {nothing} from 'lit-html';
+import style from '../css/elements/chromedash-upcoming-milestone-card.css';
+
+class ChromedashUpcomingMilestoneCard extends LitElement {
+  static styles = style;
+  static get properties() {
+    return {
+      // Assigned in schedule-apge.js, value from Django
+      showBlink: {type: Boolean},
+      starredFeatures: {type: Array},
+      templateContent: {type: Object},
+      channel: {type: Object},
+      showDates: {type: Boolean},
+      removedStatus: {type: Array},
+      deprecatedStatus: {type: Array},
+      signedIn: {type: Boolean},
+    };
+  }
+
+  /**
+   *  Returns the number of days between a and b.
+   *  @param {!Date} a
+   *  @param {!Date} b
+   *  @return {!{days: number, future: boolean}}
+   */
+  _dateDiffInDays(a, b) {
+    // Discard time and time-zone information.
+    const utc1 = Date.UTC(a.getFullYear(), a.getMonth(), a.getDate());
+    const utc2 = Date.UTC(b.getFullYear(), b.getMonth(), b.getDate());
+
+    const MS_PER_DAY = 1000 * 60 * 60 * 24;
+    const daysDiff = Math.floor((utc2 - utc1) / MS_PER_DAY);
+    return {days: Math.abs(daysDiff), future: daysDiff < 1};
+  }
+
+  _computeDate(dateStr) {
+    const opts = {/* weekday: 'short', */month: 'short', day: 'numeric'};
+    const date = new Date(dateStr);
+    return new Intl.DateTimeFormat('en-US', opts).format(date);
+  }
+
+  _fireEvent(eventName, detail) {
+    let event = new CustomEvent(eventName, {
+      bubbles: true,
+      composed: true,
+      detail,
+    });
+    this.dispatchEvent(event);
+  }
+
+  toggleStar(e) {
+    e.preventDefault();
+    e.stopPropagation();
+
+    const iconEl = e.target;
+    const featureId = Number(iconEl.dataset.featureId);
+    const newStarred = !this.starredFeatures.includes(featureId);
+
+    this._fireEvent('star-toggle-event', {
+      feature: featureId,
+      doStar: newStarred,
+    });
+  }
+
+  _computeDaysUntil(dateStr) {
+    const today = new Date();
+    const diff = this._dateDiffInDays(new Date(dateStr), today);
+    const prefix = diff.future ? 'in' : '';
+    const days = diff.days > 1 ? 's' : '';
+    const ago = !diff.future ? 'ago' : '';
+    return `${prefix} ${diff.days} day${days} ${ago}`;
+  }
+
+  _objKeys(obj) {
+    if (!obj) {
+      return [];
+    }
+    return Object.keys(obj).sort();
+  }
+
+  render() {
```

Comment:
This would probably be more clear if render() used one helper function to render the info at the top of the card, and a second helper function to render the feature list.  And, that feature list helped function might rely on a third helper function that renders one feature item along with the icons.

Reply:
Okay. This seems to be the better approach. 

Done. See [this](https://github.com/GoogleChrome/chromium-dashboard/pull/1432/commits/b77f2604c3870df414b27bfc7cb4e7026e96b161) commit.

Reply:
Looks good.